packages/peregrine/lib/talons/AuthModal/useAuthModal.js

```diff
@@ -61,14 +60,9 @@ export const useAuthModal = props => {
     }, [showMyAccount]);
 
     const handleSignOut = useCallback(async () => {
-        // After logout, reset the store to set the bearer token.
-        // https://www.apollographql.com/docs/react/networking/authentication/#reset-store-on-logout
-        await resetStore();
```

Comment:
So where do we actually clear everything out now?

Reply:
Thanks for pointing this out. This is magical so let me explain (and potentially apply a change).

When we call the `signOut` action we delete cart data from our **redux** browser persistence layer. If you throw a debugger on `history.go(0)` and inspect local storage you'll see that **apollo's** persistence still contains the `Cart` entry with the old id, however there will be no other reference. So when the page refreshes we still have the old cart entry. However, on page start the code that is responsible for initializing the cart checks **redux's** persistence for an existing id, finds nothing, and calls the `createCart` mutation. Our authLink automatically sends a bearer token (which is null because we're now a guest) with the mutation which sees us as a guest and responds with a brand spankin' new cart id. We then make a query for `getCartDetails` using this new cart id, and because I rewrote all the connection keys to store cart responses in a single location, we blow away the old cart data that was there from the authed user.

SO

TL;DR - A "proper" thing to do would be to call `deleteCacheEntry(...)` here before we refresh. :)

Reply:
> brand spankin' new cart id.

😄